packages/ra-core/src/core/useConfigureAdminRouterFromChildren.tsx

```diff
@@ -60,28 +60,27 @@ export const useConfigureAdminRouterFromChildren = (
     >(routesAndResources.resources);
 
     const [status, setStatus] = useSafeSetState<AdminRouterStatus>(() =>
-        !!getSingleChildFunction(children)
-            ? 'loading'
-            : resources.length > 0
-            ? 'ready'
-            : 'empty'
+        getStatus({
+            children,
+            resources,
```

Comment:
I suggest you use the same `routesandResources`  as in the effect, so that the reader doesn't try to understand why there is a difference